anker/src/processor.rs

```diff
@@ -369,6 +374,59 @@ fn process_change_token_swap_pool(
     anker.save(accounts.anker)
 }
 
+/// Send rewards via Wormhole from the UST reserve address to Terra.
+fn process_send_rewards(
+    program_id: &Pubkey,
+    accounts_raw: &[AccountInfo],
+    nonce: u32,
+    fee: u64,
+) -> ProgramResult {
+    let accounts = SendRewardsAccountsInfo::try_from_slice(accounts_raw)?;
+    let (solido, anker) = deserialize_anker(program_id, accounts.anker, accounts.solido)?;
+    let wormhole_transfer_args = anker.check_send_rewards(&solido, &accounts)?;
+    let ust_reserve_state =
+        spl_token::state::Account::unpack_from_slice(&accounts.from.data.borrow())?;
+    let reserve_ust_amount = MicroUst(ust_reserve_state.amount);
+    let payload = crate::wormhole::Payload::new(
+        nonce,
+        reserve_ust_amount,
+        fee,
+        anker.terra_rewards_destination.to_bytes(),
+    );
+
+    // Send UST tokens via Wormhole 🤞.
+    let reserve_seeds = [
+        accounts.anker.key.as_ref(),
+        ANKER_RESERVE_AUTHORITY,
+        &[anker.reserve_authority_bump_seed],
+    ];
+
+    invoke_signed(
+        &get_wormhole_transfer_instruction(&payload, &wormhole_transfer_args),
+        &[
+            accounts.payer.clone(),
+            accounts.config_key.clone(),
+            accounts.from.clone(),
+            accounts.mint.clone(),
+            accounts.custody_key.clone(),
+            accounts.authority_signer_key.clone(),
+            accounts.custody_signer_key.clone(),
+            accounts.bridge_config.clone(),
+            accounts.message.clone(),
+            accounts.emitter_key.clone(),
+            accounts.sequence_key.clone(),
+            accounts.fee_collector_key.clone(),
+            accounts.sysvar_clock.clone(),
+            accounts.sysvar_rent.clone(),
+            accounts.system_program.clone(),
+            accounts.wormhole_core_bridge_program_id.clone(),
+            accounts.spl_token.clone(),
+            accounts.wormhole_token_bridge_program_id.clone(),
```

Comment:
We can't make so many clones. Wonder why?! Maybe we reached some limits, and the compiler is trying to be clever.
The BPF program is all screwed up if we include this code

Reply:
What do you mean “The BPF program is all screwed up”? I don’t expect the compiler to behave any differently depending on the amount of clones.

Reply:
There are all sort of errors in the tests, e.g., I put some `msg!` to debug and the errors change. Access violations, etc.
If I put 1 account less in this function, the errors don't happen anymore.
Maybe we don't need the last account `accounts.wormhole_token_bridge_program_id.clone(),` and we can get away with it.

Reply:
I suspect something funky going on with `solana_program_test` or something like that more than rustc miscompiling anything. Maybe a size limit of some sort somewhere? Also, we had issues in the past with inexplicable errors where a clean build fixed things.

> Maybe we don't need the last account accounts.wormhole_token_bridge_program_id.clone(), and we can get away with it.

It sounds very dangerous to live that close to the edge, especially if we don’t know why the edge is there. We should understand this a bit better before we proceed. Maybe it’s possible to make a minimal example? What if we make an otherwise empty program that takes as many accounts as this instruction, and that tries to call Wormhole? Maybe we can isolate the issue a bit?